src/EnergyPlus/OutputProcessor.cc

```diff
@@ -3453,169 +3214,100 @@ void UpdateDataandReport(EnergyPlusData &state, OutputProcessor::TimeStepType co
-                                             state.dataGlobal->DayOfSimChr,
-                                             true,
-                                             state.dataEnvrn->Month,
-                                             state.dataEnvrn->DayOfMonth,
-                                             state.dataGlobal->HourOfDay,
-                                             op->TimeValue[(int)t_TimeStepTypeKey].CurMinute,
-                                             StartMinute,
-                                             state.dataEnvrn->DSTIndicator,
-                                             ScheduleManager::dayTypeNames[CurDayType]);
-                    op->LHourP = state.dataGlobal->HourOfDay;
-                    op->LStartMin = StartMinute;
-                    op->LEndMin = op->TimeValue[(int)t_TimeStepTypeKey].CurMinute;
+        if (var->freq != ReportFreq::EachCall) continue;
+        
+        if (TimePrint) {
+            if (op->LHourP != state.dataGlobal->HourOfDay || std::abs(op->LStartMin - StartMinute) > 0.001 ||
+                std::abs(op->LEndMin - op->TimeValue[(int)t_TimeStepTypeKey].CurMinute) > 0.001) {
+                int CurDayType = state.dataEnvrn->DayOfWeek;
+                if (state.dataEnvrn->HolidayIndex > 0) {
+                    CurDayType = state.dataEnvrn->HolidayIndex;
                 }
-                TimePrint = false;
-            }
-            // only time integer vars actual report as integer only is "detailed"
-            WriteNumericData(state, iVar->ReportID, *iVar->Which);
-            ++state.dataGlobal->StdOutputRecordCount;
-
-            if (rf->timeSeriesEnabled()) {
-                rf->detailedTSData[(int)t_TimeStepTypeKey].pushVariableValue(iVar->ReportID, *iVar->Which);
+                WriteTimeStampFormatData(state,
+                                         state.files.eso,
+                                         ReportFreq::EachCall,
+                                         op->freqStampReportNums[(int)ReportFreq::TimeStep],
+                                         state.dataGlobal->DayOfSimChr,
+                                         true,
+                                         state.dataEnvrn->Month,
+                                         state.dataEnvrn->DayOfMonth,
+                                         state.dataGlobal->HourOfDay,
+                                         op->TimeValue[(int)t_TimeStepTypeKey].CurMinute,
+                                         StartMinute,
+                                         state.dataEnvrn->DSTIndicator,
+                                         ScheduleManager::dayTypeNames[CurDayType]);
+                op->LHourP = state.dataGlobal->HourOfDay;
+                op->LStartMin = StartMinute;
+                op->LEndMin = op->TimeValue[(int)t_TimeStepTypeKey].CurMinute;
             }
+            TimePrint = false;
```

Comment:
Not sure why git makes these changes look so interleaves but the top level change is that there were two essentially identical loops, one for real variables and one for int variables and now there is just one.